src/adapter/src/coord/ddl.rs

```diff
@@ -497,18 +515,82 @@ impl Coordinator {
             catalog,
             active_conns,
             controller,
+            cluster_replica_statuses,
             ..
         } = self;
         let catalog = Arc::make_mut(catalog);
         let conn = conn_id.map(|id| active_conns.get(id).expect("connection must exist"));
 
         let TransactionResult {
-            builtin_table_updates,
+            mut builtin_table_updates,
             audit_events,
         } = catalog
             .transact(Some(&mut *controller.storage), oracle_write_ts, conn, ops)
             .await?;
 
+        // Update in-memory cluster replica statuses.
+        // TODO(jkosh44) All the builtin table updates should be handled as a builtin source
+        // updates elsewhere.
+        for (cluster_id, replica_id) in &cluster_replicas_to_drop {
+            let replica_statuses =
+                cluster_replica_statuses.remove_cluster_replica_statuses(cluster_id, replica_id);
+            for (process_id, status) in replica_statuses {
+                let builtin_table_update = catalog.state().pack_cluster_replica_status_update(
+                    *replica_id,
+                    process_id,
+                    &status,
+                    -1,
+                );
+                builtin_table_updates.push(builtin_table_update);
+            }
+        }
+        for cluster_id in &clusters_to_drop {
+            let cluster_statuses = cluster_replica_statuses.remove_cluster_statuses(cluster_id);
+            for (replica_id, replica_statuses) in cluster_statuses {
+                for (process_id, status) in replica_statuses {
+                    let builtin_table_update = catalog
+                        .state()
+                        .pack_cluster_replica_status_update(replica_id, process_id, &status, -1);
+                    builtin_table_updates.push(builtin_table_update);
+                }
+            }
+        }
+        for cluster_id in clusters_to_create {
+            cluster_replica_statuses.initialize_cluster_statuses(cluster_id);
+            for (replica_id, replica_statuses) in
+                cluster_replica_statuses.get_cluster_statuses(cluster_id)
+            {
+                for (process_id, status) in replica_statuses {
+                    let builtin_table_update = catalog.state().pack_cluster_replica_status_update(
+                        *replica_id,
+                        *process_id,
+                        status,
+                        1,
+                    );
+                    builtin_table_updates.push(builtin_table_update);
+                }
+            }
+        }
+        for (cluster_id, replica_id, num_processes) in cluster_replicas_to_create {
+            cluster_replica_statuses.initialize_cluster_replica_statuses(
+                cluster_id,
+                replica_id,
+                num_processes,
+                to_datetime((catalog.config().now)()),
```

Comment:
If there's multiple replicas in the vec, they could all get different timestamps? Is this a problem?

Reply:
The previous behavior was actually that each individual process got it's own timestamp, and I updated it so every process in the same replica gets the same timestamp. I think it's worth taking it one step further and assigning every process in every replica the same timestamp. I just pushed that update.